scalalib/src/ScalaModule.scala

```diff
@@ -213,7 +218,7 @@ trait ScalaModule extends JavaModule { outer =>
       ) match{
         case true =>
           val inputPath =
-            if (isDottyOrScala3(scalaVersion())) javadocDir / '_site
+            if (isDottyOrScala3(scalaVersion())&& !useScaladocInScala3(scalaVersion())) javadocDir / '_site
             else javadocDir
```

Comment:
would this still keep static files in scaladoc3? We already use `javadocDir` to consolidate all static files (so that we can support multiple effective site roots), so I think that because scaladoc3 no longer targets a `_site` subdirectory, it could lead into strange file-overwriting behavior. 

When I worked on this, I needed to create a separate directory to use as the site root. See [here](https://github.com/com-lihaoyi/mill/pull/1190/files#diff-41bb9fa9050194a7a0758e000dec4675497d8d907bbfe210bb9e711be10b55adR215-R219).